dashboard/app/models/script.rb

```diff
@@ -1361,7 +1367,13 @@ def self.clear_cache
   end
 
   def localized_title
-    I18n.t "data.script.name.#{name}.title"
+    title = I18n.t "data.script.name.#{name}.title"
+    has_prefix = unit_group&.has_numbered_units
+    return title unless has_prefix
+
+    position = unit_group_units&.first&.position
```

Comment:
Do these require a DB query each time? Or do they get cached somehow? Not sure if it's relevant here but thought I'd ask.